drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -66,15 +66,15 @@ static int pd_intel_adsp_init(const struct device *dev)
 
 #define DT_DRV_COMPAT intel_adsp_power_domain
 
-#define POWER_DOMAIN_DEVICE(id)                                                                    \
-	static struct pg_registers pd_pg_reg##id = {                                               \
-		.wr_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), lps)) + PWRCTL_OFFSET,       \
-		.rd_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), lps)) + PWRSTS_OFFSET,       \
-		.SPA_bit = DT_INST_PROP(id, bit_position),                                         \
-		.CPA_bit = DT_INST_PROP(id, bit_position),                                         \
-	};                                                                                         \
-	PM_DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_pm_action);                                     \
-	DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_init, PM_DEVICE_DT_INST_GET(id),                   \
+#define POWER_DOMAIN_DEVICE(id)
+	static struct pg_registers pd_pg_reg##id = {
+		.wr_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), dfpmcch)) + PWRCTL_OFFSET,
+		.rd_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), dfpmcch)) + PWRSTS_OFFSET,
+		.SPA_bit = DT_INST_PROP(id, bit_position),
+		.CPA_bit = DT_INST_PROP(id, bit_position),
+	};
+	PM_DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_pm_action);
+	DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_init, PM_DEVICE_DT_INST_GET(id),
 			      &pd_pg_reg##id, NULL, POST_KERNEL, 75, NULL);
 
```

Comment:
How this work ? `DT_INST_FOREACH_STATUS_OKAY` expands to something that iterates N types calling the given macro with the instance value. This way `PM_DEVICE_DT_INST_DEFINE` and `DEVICE_DT_INST_DEFINE` will never be called.

Reply:
As you see I just have cosmetic style changes here. 
@kfrydryx could you please answer @ceolin question as it was done by you (https://github.com/zephyrproject-rtos/zephyr/pull/45475)? Maybe it wasn't noticed before...

Reply:
updated

Reply:
@ceolin  please re-review and resolve the conversation if it is o.k for you ;)

Reply:
it was not a cosmetic change, the backslash was removed breaking the functionality :)